recipes/seadex-essentials/all/conanfile.py

```diff
@@ -0,0 +1,104 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.files import export_conandata_patches, get, copy
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+required_conan_version = ">=1.59"
+
+class SeadexEssentialsConan(ConanFile):
+    name = "seadex-essentials"
+    description = "essentials is a small c++ library that offers very basic capabilities for applications and libraries."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://essentials.seadex.de/"
+    topics = ("utility", "C++", "library")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "spdlog/*:header_only": True
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "8.3",
+            "clang": "12",
+            "Visual Studio": "16",
+            "msvc": "192",
+            "apple-clang": "10"
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            self.options.shared = False
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+        if self.settings.compiler == "clang" or self.settings.compiler == "apple-clang":
+            self.settings.compiler.cppstd = self._min_cppstd
```

Comment:
I don't think you want to actively modify the `settings.compiler.cppstd`, but instead check it in the validate method, as you already do.

```suggestion
```